plugins/catalog-backend/src/service/DefaultLocationService.ts

```diff
@@ -54,6 +54,38 @@ export class DefaultLocationService implements LocationService {
     return this.store.deleteLocation(id);
   }
 
+  private async processEntities(
+    unprocessedEntities: DeferredEntity[],
+  ): Promise<Entity[]> {
+    const entities: Entity[] = [];
+    while (unprocessedEntities.length) {
+      const currentEntity = unprocessedEntities.pop();
+      if (!currentEntity) {
+        continue;
+      }
+      const processed = await this.orchestrator.process({
+        entity: currentEntity.entity,
+        state: {}, // we process without the existing cache
+      });
+
+      if (processed.ok) {
+        const { metadata, kind } = processed.completedEntity;
+        if (
+          entities.some(
+            e => e.metadata.name === metadata.name && e.kind === kind,
```

Comment:
This should compare the `metadata.namespace` too. The entire triplet is what uniqueness is based upon.

Feel free to use `stringifyEntityRef` from `@backstage/catalog-model` on both sides and compare those strings. That deals with some nuances such as case insensitivity and default namespace.

Reply:
Ok sweet. Thanks for that!